### DISCUSSION TITLE
Implementing with "document" as a target in a Gatsby project using useEffect

### DISCUSSION THREAD

**Discussion by adamatronix** [pmndrs/use-gesture#564] [Q&A] [answered]
Hi,

I checked out the documentation where in past versions you were able to do something like:
` 
const wheel = useWheel({...},{ target: document })
useEffect(wheel, [wheel])`

However now it says the hook will do the clean up itself. Basically I need to check when "document" or the dom has loaded as during a Gatsby build non is available. I would like to run it in a useEffect.

Thanks

**Comment by dbismut** [comment, marked as answer]
What would you want to run in `useEffect`? Not sure I understand, that doesn't make much sense to me sorry.

**Reply by adamatronix** [reply]
When the component is mounted the useWheel is called and looks for the target to add their listeners I assume. But "document" is undefined. I would like to initialise the useWheel hook only when "document" is defined.

**Reply by adamatronix** [reply]
Sorry I guess just placing a conditional works. But was wondering if it could be used within the component lifecycle.

**Reply by dbismut** [reply]
Hm but listeners are added in an effect hook internally so I'm not sure what's going on. I'm on my phone I'll have a look when I'm back home.

**Reply by adamatronix** [reply]
Sure np.

**Reply by dbismut** [reply]
Yeah one way to solve this would be to have target accept a function (like `{ target: () => document }`). Until then, you can just use the usual `typeof document !== 'undefined' ? document : undefined` as I think you mentioned.

Also note that the previous version of use-gesture wouldn't have solve this. The sheer fact that your code mentions a non defined object in its render function at build time makes the SSR build crash.

**Reply by adamatronix** [reply]
Cool. Yea ended up just adding a conditional around the hook.

**Reply by dbismut** [reply]
Note that you shouldn't write a conditional _around_ the hook. That goes against the Rules.
